File: FITCCRS2.Model/Tim.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace FITCCRS2.Model
{
    public class Tim
    {
        public int TimId { get; set; }
        public string Naziv { get; set; }
        public int? BrojClanova { get; set; }
        public int? TakmicenjeId { get; set; }
        public int? UserId { get; set; }
        public string Username { get; set; }

        public virtual Takmicenje Takmicenje { get; set; }
        public virtual ICollection<Projekat> Projekats { get; set; }
    }
}

File: FITCC.UI/admin_fitcc/lib/models/tim.dart
import 'package:admin_fitcc/models/projekat.dart';
class Tim {
  late int timId;
  late String naziv;
  late int brojClanova;
  late int takmicenjeId;
  late int userId;
  late String username;
  List<Projekat> projekats = [];
  
  Tim({
    required this.timId,
    required this.naziv,
    required this.brojClanova,
    required this.takmicenjeId,
    required this.userId,
    required this.username,
    List<Projekat>? projekats,
  }) {
    if (projekats != null) {
      this.projekats = projekats;
    }
  }
   factory Tim.fromJson(Map<String, dynamic> json) {
    var projekatsJson = json['projekats'] as List<dynamic>;
    var projekatsList = projekatsJson.map((projekatJson) {
      return Projekat.fromJson(projekatJson as Map<String, dynamic>);
    }).toList();

    return Tim(
      timId: json['timId'] as int,
      naziv: json['naziv'] as String,
      brojClanova: json['brojClanova'] as int,
      takmicenjeId: json['takmicenjeId'] as int,
      userId: json['userId'] as int,
      username: json['username'] as String,
      projekats: projekatsList,
    );
  }
}
